packages/core/mesh/edge-client/src/edge-identity.ts

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { invariant } from '@dxos/invariant';
+import { schema } from '@dxos/protocols/proto';
+import { type Presentation } from '@dxos/protocols/proto/dxos/halo/credentials';
+
+export interface EdgeIdentity {
+  peerKey: string;
+  identityKey: string;
+  /**
+   * Returns credential presentation issued by the identity key.
+   * Presentation must have the provided challenge.
+   * Presentation may include ServiceAccess credentials.
+   */
+  presentCredentials({ challenge }: { challenge: Uint8Array }): Promise<Presentation>;
+}
+
+export const handleAuthChallenge = async (failedResponse: Response, identity: EdgeIdentity): Promise<Uint8Array> => {
```

Comment:
I suggest separating this into `extractChallenge` and `generateAuthPresentation(identity, challenge)`

Reply:
If these two methods are **always** going to be called one after another, is there a point?
I'd do this if one of the methods needs to be reused in other context.